### PR TITLE
Add Cave Story 3D

### PR DESCRIPTION
**Pull request by TudorykMakNark**
Adds Cave Story 3D.

*Automatically generated by PR builder*

### CODE REVIEW

File: games/cave-story-3d/game.dat [hunk 1]
```diff
@@ -0,0 +1,19 @@
+title = "Cave Story 3D"
+description = "Remake of Original \"Cave Story\" for 3DS, a Metroidvania game style."
```
Comment (Leo626) [inline games/cave-story-3d/game.dat]: Follow the wikipedia style like this.

`Cave Story 3D, known as \"Doukutsu Monogatari 3D\" in Japan, is a metroidvania platform-adventure video game developed by Nicalis and published by NIS America, and a 3D remake of Cave Story. It features updated 3D graphics, 3 game modes, and remastered music.`

File: games/cave-story-3d/game.dat [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+[[releases]]
+title = "0004000000084C00"
+region = "jpn"
+release_date = "2012-07-26"
+
+[[releases]]
+title = "000400000004D200"
+region = "eur"
+release_date = "2012-07-26"
```
Comment (Leo626) [inline games/cave-story-3d/game.dat]: Wrong date. Should be `2011-11-11`.

File: games/cave-story-3d/game.dat [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+[[releases]]
+title = "000400000004A100"
+region = "usa"
+release_date = "2012-07-26"
```
Comment (Leo626) [inline games/cave-story-3d/game.dat]: Wrong date. Should be `2011-11-08`.

**Comment by Leo626** [comment]
@TudorykMakNark Note that you can make changes to your [forked repository](https://github.com/TudorykMakNark/citra-games-wiki/tree/3b412456519e0ac8477506d41cf6ab248e2b000b/games/cave-story-3d) instead of opening a new pull request.

**Comment by TudorykMakNark** [comment]
@Leo626 I don't know if I have made the right things, I'm not familiarized with Github, sorry, but in some way I made all changes you request

**Comment by Leo626** [comment]
@TudorykMakNark The screenshots show the mouse cursor. Disable Show cursor in ShareX. Also the box art is the European one. A high quality source for USA one can be found [here](https://art.gametdb.com/3ds/coverHQ/US/ACVE.jpg).

**Comment by TudorykMakNark** [comment]
@Leo626 I think now is all right

**Comment by TudorykMakNark** [comment]
now I have hopes all is right! 700 tries later haha sry

**Pull request merged**